### PR TITLE
Add Xeus-clang-repl documentation

### PR DESCRIPTION
**Pull request by Krishna-13-cyber**
This PR adds doxygen and sphinx documentation for xeus-clang-repl.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,16 +43,16 @@ configure_file (
 ################
 
 set(xeus_REQUIRED_VERSION 1.0.0)
-set(cppzmq_REQUIRED_VERSION 4.3.0)
```
Comment (vgvassilev) [inline CMakeLists.txt]: Can we bring this back?

File: docs/DevelopersDocumentation.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Developers Documentation
+-------------------------
```
Comment (vgvassilev) [inline docs/DevelopersDocumentation.rst]: All underscoring lines seem off by one.

File: docs/FAQ.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+FAQ 
+*****
```
Comment (vgvassilev) [inline docs/FAQ.rst]:
```suggestion
***
```

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    "loader": {"load": ["[tex]/physics"]},
+    "tex": {"packages": {"[+]": ["physics"]}},
+}
+
+import os
+XEUS_CLANG_REPL_ROOT = os.path.abspath('..')
+html_extra_path = [XEUS_CLANG_REPL_ROOT + '/build/docs/']
+
+import subprocess
+command = 'mkdir {0}/build; cd {0}/build; cmake ../ -DClang_DIR=/usr/lib/llvm-14/build/lib/cmake/clang\
```
Comment (vgvassilev) [inline docs/conf.py:53]: Can we avoid hardcoding this `llvm-14` version.

Comment (Krishna-13-cyber) [reply docs/conf.py:53]: We are fetching llvm-14(dev, tools) for the doc builds to take place. Therefore we can keep the specific llvm-versions for cmake checks.

**Review by vgvassilev — APPROVED** [review]
LGTM. Can you squash the changes into a single commit with a good comment?

**Pull request merged**